content/blog/2024-12-22-az-affinity-strategy.md

```diff
@@ -0,0 +1,166 @@
+**Example:**
+Consider a cluster with three nodes, primary and two replicas. Each node located in different availability zone. The client locate in az-2 as replica-1. 
+
+    **With ```PERFER_REPLICA``` strategy**:
+    In this case, the client will read commands from any replica that is available. It may be located in a different AZ as shown below, and the average latency is, for example, 800 microseconds.
+
+    ![PERFER_REPLICA Read strategy latency example](/assets/media/pictures/PERFER_REPLICA_strategy.png)
+
+
+    **With ```AZ_AFFINITY``` strategy**:
+    In this case, the client will read commands from a replica in the same client's AZ and the average latency is, for example, about 300 microseconds.
+
+    ![AZ_AFFINITY Read strategy latency example](/assets/media/pictures/AZ_AFFINITY_strategy.png)
+
+## Configuring AZ Affinity Connections with GLIDE
+
+Setting up AZ affinity routing in GLIDE is simple, allowing you to leverage its full potential with just a few configuration steps. Let’s walk through the steps to enable this feature in your application.
+
+### Steps to Set Up AZ Affinity Routing in GLIDE
+
+1. Configure Valkey nodes availability zone
+    Assign each Valkey node to a specific AZ based on its physical or virtual location within the Cloud provider's region. 
+    The initial configuration must to be done with a separate management client on node initialization, as the clients gets the info from the replicas on the first reconnect. 
+    In some managed services like Amazon ElastiCache, this mapping is configured automatically and this step is not required. 
+
+    For each node, run the following command and change the AZ and routing address as appropriate:
+
+    **Python:**
+    ```python
+    client.config_set({"availability-zone": az}, 
+                        route=ByAddressRoute(host="address.example.com", port=6379))
+    ```
+    **Java:**
+    ```Java
+    client.configSet(Map.of("availability-zone", az), new ByAddressRoute("address.example.com", 6379))
+    ```
+    **Node.js:**
+    ```javascript
+    client.configSet({"availability-zone": az}, { route: {type: "routeByAddress", host:"address.example.com", port:6379}})
+    ```
+
+2. Configure GLIDE with AZ-Specific Targeting
+    Here are Python, Java, and Node.JS examples showing how to set up an AZ affinity client that directs calls to nodes in the same AZ as the client.
+    ```python
+    from glide import (
+        GlideClient,
+        GlideClientConfiguration,
+        NodeAddress,
+        ReadFrom
+    )
+
+    # Determine the client's AZ (this could be fetched from the cloud provider's metadata service)
+    client_az = 'AZ-a'
```

Comment:
```suggestion
    client_az = 'us-east-1a'
```
You inconsistently use `us-east-1a` as an example and `AZ-a`, but the former is clearer imo. I think we can be AWS centric since we wrote the blog, but explain how it works generally.